bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java

```diff
@@ -369,7 +369,7 @@ private String getPageLinkURL(@NotNull Page page) {
             redirectCandidates.add(page.getPath());
             while (result != null && StringUtils
                     .isNotEmpty((redirectTarget = result.getProperties().get(PageImpl.PN_REDIRECT_TARGET, String.class)))) {
-                result = pageManager.getPage(redirectTarget);
+                result = page.getPageManager().getPage(redirectTarget);
```

Comment:
to be on the safe side, I would add a null check for `page.getPageManager()`

Reply:
Added extra null-check in fa8ba7b